pyclesperanto_prototype/_tier9/_statistics_of_labelled_pixels.py

```diff
@@ -92,3 +106,284 @@ def standard_deviation_intensity(region, intensities):
 
     return props
 
+def _statistics_of_labelled_pixels_gpu(intensity_image : Image = None, label_image : Image = None, measure_shape : bool = True):
+    """
+    This method will replate statistics_of_labelled_pixels(use_gpu=True) at some point.
+
+    Parameters
+    ----------
+    intensity_image
+    label_image
+    measure_shape
+
+    Returns
+    -------
+
+    """
+    from .._tier0 import create
+    from .._tier0 import create_like
+    from .._tier1 import set
+    from .._tier1 import sum_y_projection
+    from .._tier1 import minimum_y_projection
+    from .._tier1 import maximum_y_projection
+    from .._tier1 import crop
+    from .._tier1 import paste
+    from .._tier1 import divide_images
+    from .._tier1 import set_plane
+    from .._tier0 import pull_zyx
+    from .._tier1 import power
+    from .._tier2 import maximum_of_all_pixels
+    import numpy as np
+
+    # check if label image and intensity image a properly set
+    if intensity_image is None and label_image is None:
+        raise ValueError("Either intensity image or labels must be set")
+    if label_image is None:
+        warn("No label image provided. All pixels will be analysed in one group.")
+        label_image = create_like(intensity_image)
+        set(label_image, 1) # all pixels belong to label 1
+    if intensity_image is None:
+        intensity_image = label_image
+
+    # when copying measurements, we skip the first column, as we don't analyse background
+    measurements_start_x = 1
+
+    # determine size of temporary images
+    num_labels = int(maximum_of_all_pixels(label_image)) + measurements_start_x
+    height = label_image.shape[-2]
+    if len(label_image.shape) == 3:
+        depth = label_image.shape[0]
+    else:
+        depth = 1
+    num_dimensions = len(label_image.shape)
+
+    # results go here
+    region_props = {}
+
+    # intermediate results are stored in this image
+    sum_per_label_image = create([16, height, num_labels])
+    set(sum_per_label_image, 0)
+
+    # before determining min/max, set initial values to very high and very low values
+    min_value = np.finfo(np.float32).min
+    max_value = np.finfo(np.float32).max
+    set_plane(sum_per_label_image, 8, max_value)
+    set_plane(sum_per_label_image, 9, min_value)
+    set_plane(sum_per_label_image, 10, max_value)
+    set_plane(sum_per_label_image, 11, min_value)
+    set_plane(sum_per_label_image, 12, max_value)
+    set_plane(sum_per_label_image, 13, min_value)
+    set_plane(sum_per_label_image, 14, max_value)
+    set_plane(sum_per_label_image, 15, min_value)
```

Comment:
Can you explain these lines? They are at the very least confusing! Not least because the planes 0-7 aren't touched for some reason?

Reply:
There is a comment two lines above explaining it. Furthermore, all lines (also 0-7) are touched, as they are set to 0. I'll rename the variable of that image, because the "sum" might be misleading.

https://github.com/clEsperanto/pyclesperanto_prototype/pull/79/files/4f6dd5a3e2f68344af6e788f15363e657cad9edc#diff-1c45f4e441422600bcfaa0213495cd24820ec27589879353d92de7b1ff911200R163-R167